steampy/client.py

```diff
@@ -311,10 +316,77 @@ def fetch_price(self, item_hash_name: str, game: GameOptions, currency: str = Cu
             raise TooManyRequests("You can fetch maximum 20 prices in 60s period")
         return response.json()
 
+    @login_required
+    def get_my_market_listings(self) -> dict:
+        response = self._session.get("%s/market" % SteamClient.COMMUNITY_URL)
+        if response.status_code == 200:
+            assets_descriptions = json.loads(text_between(response.text, "var g_rgAssets = ", ";"))
+            listing_id_to_assets_address = get_listing_id_to_assets_address_from_html(response.text)
+            listings = get_market_listings_from_html(response.text)
+            listings = merge_items_with_descriptions_from_listing(listings, listing_id_to_assets_address,
+                                                                  assets_descriptions)
+            return listings
+        else:
+            raise FetchException("Steam market page return a %s" % response.status_code)
+
+    @login_required
+    def create_sell_listing(self, assetid: str, game: GameOptions, money_to_receive: int) -> dict:
+        data = {
+            "assetid": assetid,
+            "sessionid": self._get_session_id(),
+            "contextid": game.context_id,
+            "appid": game.app_id,
+            "amount": 1,
+            "price": money_to_receive
+        }
+        headers = {'Referer': "http://steamcommunity.com/profiles/%s/inventory" % self.steam_guard['steamid']}
+        response = self._session.post("https://steamcommunity.com/market/sellitem/", data, headers=headers).json()
+        if response.get("needs_mobile_confirmation"):
+            return self._confirm_sell_listing(assetid)
+        return response
+
+    @login_required
+    def create_buy_order(self, market_name: str, price_single_item: int, quantity: int, game: GameOptions,
+                         currency: Currency = Currency.USD) -> dict:
+        data = {
+            "sessionid": self._get_session_id(),
+            "currency": currency.value,
+            "appid": game.app_id,
+            "market_hash_name": market_name,
+            "price_total": price_single_item * quantity,
+            "quantity": quantity
+        }
+        headers = {'Referer': "http://steamcommunity.com/market/listings/%s/%s" % (game.app_id, market_name)}
+        response = self._session.post("https://steamcommunity.com/market/createbuyorder/", data, headers=headers)
+        response = response.json()
+        return response
+
```

Comment:
Could you check that an raise exception with the text from str_error with added messages that it can also be due to differwnt currency?